spring-cloud-dataflow-server-core/src/main/resources/META-INF/spring.factories

```diff
@@ -5,7 +5,8 @@ org.springframework.boot.env.EnvironmentPostProcessor=\
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
   org.springframework.cloud.dataflow.server.config.DataFlowServerAutoConfiguration,\
   org.springframework.cloud.dataflow.server.config.DataFlowControllerAutoConfiguration, \
-  org.springframework.cloud.dataflow.server.config.SpringDocAutoConfiguration
+  org.springframework.cloud.dataflow.server.config.SpringDocAutoConfiguration, \
+  org.springframework.cloud.dataflow.server.config.DataflowAsyncConfiguration
```

Comment:
Lets simplify this by moving this config from auto-config to being imported by `org.springframework.cloud.dataflow.server.config.DataFlowServerConfiguration`. 

Also, let's leverage the Spring Boot [task executor builder](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.task-execution-and-scheduling) to avoid having to create another set of config props. I envision something like this: 
```java
@Configuration(proxyBeanMethods = false)
@ConditionalOnProperty(prefix = ASYNC_PREFIX, name = "enabled")
@EnableAsync
class DataflowAsyncConfiguration implements AsyncConfigurer {

	public static final String ASYNC_PREFIX = DataFlowPropertyKeys.PREFIX + "task.cleanup.async";

	@Bean(name = "taskCleanupExecutor")
	Executor taskCleanupExecutor(TaskExecutorBuilder taskExecutorBuilder) {
		return taskExecutorBuilder.threadNamePrefix("TaskCleanup-").build();
	}

	@Override
	public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
		return (throwable, method, objects) -> logger.error("Exception thrown in @Async Method " + method.getName(),
				throwable);
	}
}
```
We can add SCDF specific props later if we decide they are needed.

Reply:
This is awesome! We can refactor it like this!

Reply:
Refactored as requested 😁